packages/relay-compiler/bin/RelayCompilerMain.js

```diff
@@ -333,17 +377,16 @@ function getRelayFileWriter(
   }: WriteFilesOptions) => {
     let persistQuery;
     let queryMap;
-    if (persistedQueryPath != null) {
+    if (persistFunction != null || persistedQueryPath != null) {
       queryMap = new Map();
-      persistQuery = (text: string) => {
-        const hasher = crypto.createHash('md5');
-        hasher.update(text);
-        const id = hasher.digest('hex');
+      const persistImplmentation = persistFunction || defaultPersistFunction;
+      persistQuery = async (text: string) => {
+        const id = await persistImplmentation(text);
         queryMap.set(id, text);
-        return Promise.resolve(id);
+        return id;
       };
     }
-    const results = RelayFileWriter.writeAll({
+    const results = await RelayFileWriter.writeAll({
```

Comment:
So making `persistQuery` a promise is automatically making this function return a promise too? Nice.

Reply:
Not quite: RelayFileWriter.writeAll was already returning a promise, this is just changing to use async/await instead of directly returning the promise

Reply:
Ahh, ok yeah that makes more sense. Thanks 👍